tests/python/pants_test/backend/jvm/subsystems/test_jvm_platform.py

```diff
@@ -201,6 +202,52 @@ def test_synthetic_target_runtime_platform_lookup(self):
             == "default-platform"
         )
 
+    def test_strict_usage(self):
+        init_subsystem(
+            JvmPlatform,
+            options={
+                "jvm-platform": {
+                    "platforms": {
+                        "default-platform": {"target": "9"},
+                        "strict-platform": {"target": "8", "strict": True},
+                    },
+                    "default_platform": "default-platform",
+                    "default_runtime_platform": None,
+                }
+            },
+        )
+        instance = JvmPlatform.global_instance()
+        strict_platform = instance.get_platform_by_name("strict-platform")
+        default_platform = instance.default_platform
+        # TODO maybe this should use the runtime platform
+        assert instance._preferred_jvm_distribution_args([]) == {
+            "jdk": False,
+        }
+        assert JvmPlatform._preferred_jvm_distribution_args([default_platform]) == {
+            "minimum_version": Revision.lenient("9.0.0"),
+            "maximum_version": None,
+            "jdk": False,
+        }
+        assert JvmPlatform._preferred_jvm_distribution_args([default_platform], strict=True) == {
+            "minimum_version": Revision.lenient("9.0.0"),
+            "maximum_version": Revision.lenient("9.0.9999"),
+            "jdk": False,
+        }
+        assert instance._preferred_jvm_distribution_args([strict_platform]) == {
+            "minimum_version": Revision.lenient("1.8.0"),
+            "maximum_version": Revision.lenient("1.8.9999"),
+            "jdk": False,
+        }
+        assert instance._preferred_jvm_distribution_args([strict_platform], strict=False) == {
+            "minimum_version": Revision.lenient("1.8.0"),
+            "maximum_version": None,
+            "jdk": False,
+        }
+        # strict/non-strict platform requested together--not strict (maybe should be strict?)
```

Comment:
Before I land this, I think I want to resolve this question.

Reply:
I think the way to resolve may be to kill the fake? plural. My quick tracing of the public `preferred_jvm_distribution` method shows 0 calls with sets of size >1.

Reply:
JvmTask's `preferred_jvm_distribution` calls into JvmPlatform's with a possibly plural list of platforms.

That is called from 
* benchmark_run
* javadoc_gen
* junit_run (this should already be collated by platform though)
* scala_repl
* scaladoc_gen
All of those apart from junit_run may have targets with different platform configurations.

Reply:
Ah, via `pants.backend.jvm.tasks.jvm_task.JvmTask.preferred_jvm_distribution_for_targets` missed that one call site. Ok, noodling then...

Reply:
I think, after considering for a bit that the current behavior is wrong. I think if one platform is strict, then we should use a distro that matches the strict one, unless it conflicts with the others min versions. If multiple are strict and their releases don't match, then it's an error.

Reply:
lol. Yep. I'll update this tomorrow to reflect that understanding.